_ext/repository.rb

```diff
@@ -278,6 +276,22 @@ def execute(site)
 
         end
 
+        def parse_link_headers(response)
+          links = Hash.new
+          if response.headers[:link].nil?
+            return links
+          end
+          parts = response.headers[:link].split(', ')
+
+          parts.each do |part, index|
+            section = part.split(';')
+            url = section[0][/<(.*)>/,1]
```

Comment:
we should check length for section here. otherwise we'll get `section[0]` or `section[1]` as `nil`

Reply:
I think in such a case we would have invalid `link` header according to the [spec](https://tools.ietf.org/html/rfc5988) or am I missing something? We can always be defensive though.

Reply:
specification of link header helps. Thanks